### PR TITLE
Fix problem about ElasticSearch container

### LINKED ISSUE #881: Elasticsearch storage driver
@hacpai 
Hi I'm waiting for a long time for this driver, and I'm started to test it right away and found some critical issues. 
Elasticsearche storage driver works OK, when sending data to default url "http://localhost:9200", but when i trying to send to remote elasticsearch I'm getting error. 

When I'm trying to write to the existing Elasticsearch:
```
core@77cc656c-1adc-473c-ae2a-82fb3c836ede ~/bin $ ncat --telnet 10.128.129.121 9200
### everything is ok.
core@77cc656c-1adc-473c-ae2a-82fb3c836ede ~/bin $ bin/cadvisor -storage_driver="elasticsearch" -port=8099 -alsologtostderr=true -storage_driver_es_host="http://10.128.129.121:9200"
panic: no Elasticsearch node available

goroutine 1 [running]:
github.com/google/cadvisor/storage/elasticsearch.New(0xc208102780, 0x24, 0xb66370, 0x8, 0xb45230, 0x5, 0x7ffd14dc5da7, 0x1a, 0x0, 0x0, ...)
	/home/tpost/go/src/github.com/google/cadvisor/storage/elasticsearch/elasticsearch.go:104 +0x1a6
main.NewMemoryStorage(0x7ffd14dc5d60, 0xd, 0xbc1830, 0x0, 0x0)
	/home/tpost/go/src/github.com/google/cadvisor/storagedriver.go:111 +0x716
main.main()
	/home/tpost/go/src/github.com/google/cadvisor/cadvisor.go:63 +0x18a

goroutine 5 [chan receive]:
github.com/golang/glog.(*loggingT).flushDaemon(0x11ffe80)
	/home/tpost/go/src/github.com/google/cadvisor/Godeps/_workspace/src/github.com/golang/glog/glog.go:839 +0x78
created by github.com/golang/glog.init·1
	/home/tpost/go/src/github.com/google/cadvisor/Godeps/_workspace/src/github.com/golang/glog/glog.go:406 +0x2a7

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:2232 +0x1

goroutine 14 [IO wait]:
net.(*pollDesc).Wait(0xc208011790, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc208011790, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).Read(0xc208011730, 0xc20800f000, 0x1000, 0x1000, 0x0, 0x7fe991287fb0, 0xc208080760)
	/usr/local/go/src/net/fd_unix.go:242 +0x40f
net.(*conn).Read(0xc20803a2d0, 0xc20800f000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:121 +0xdc
net/http.noteEOFReader.Read(0x7fe99128a278, 0xc20803a2d0, 0xc20807b918, 0xc20800f000, 0x1000, 0x1000, 0x9ba7c0, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:1270 +0x6e
net/http.(*noteEOFReader).Read(0xc20801fe00, 0xc20800f000, 0x1000, 0x1000, 0xc208012000, 0x0, 0x0)
	<autogenerated>:125 +0xd4
bufio.(*Reader).fill(0xc208038b40)
	/usr/local/go/src/bufio/bufio.go:97 +0x1ce
bufio.(*Reader).Peek(0xc208038b40, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/bufio/bufio.go:132 +0xf0
net/http.(*persistConn).readLoop(0xc20807b8c0)
	/usr/local/go/src/net/http/transport.go:842 +0xa4
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0xc9f

goroutine 11 [syscall]:
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:21 +0x1f
created by os/signal.init·1
	/usr/local/go/src/os/signal/signal_unix.go:27 +0x35

goroutine 15 [select]:
net/http.(*persistConn).writeLoop(0xc20807b8c0)
	/usr/local/go/src/net/http/transport.go:945 +0x41d
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0xcbc
```

When I'm trying to write to unexciting elasticsearch:
```
core@77cc656c-1adc-473c-ae2a-82fb3c836ede ~/bin $ ncat --telnet 10.128.129.12 9200          
Ncat: Connection refused.


core@77cc656c-1adc-473c-ae2a-82fb3c836ede ~/bin $ bin/cadvisor -storage_driver="elasticsearch" -port=8099 -alsologtostderr=true -storage_driver_es_host="http://10.128.129.12:9200"
panic: no Elasticsearch node available

goroutine 1 [running]:
github.com/google/cadvisor/storage/elasticsearch.New(0xc208102780, 0x24, 0xb66370, 0x8, 0xb45230, 0x5, 0x7ffec742bda8, 0x19, 0x0, 0x0, ...)
	/home/tpost/go/src/github.com/google/cadvisor/storage/elasticsearch/elasticsearch.go:104 +0x1a6
main.NewMemoryStorage(0x7ffec742bd61, 0xd, 0xbc1830, 0x0, 0x0)
	/home/tpost/go/src/github.com/google/cadvisor/storagedriver.go:111 +0x716
main.main()
	/home/tpost/go/src/github.com/google/cadvisor/cadvisor.go:63 +0x18a

goroutine 5 [chan receive]:
github.com/golang/glog.(*loggingT).flushDaemon(0x11ffe80)
	/home/tpost/go/src/github.com/google/cadvisor/Godeps/_workspace/src/github.com/golang/glog/glog.go:839 +0x78
created by github.com/golang/glog.init·1
	/home/tpost/go/src/github.com/google/cadvisor/Godeps/_workspace/src/github.com/golang/glog/glog.go:406 +0x2a7

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:2232 +0x1

goroutine 11 [syscall]:
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:21 +0x1f
created by os/signal.init·1
	/usr/local/go/src/os/signal/signal_unix.go:27 +0x35
```

Also I have a reason to believe, that no matter which Elasticsearch URL I set, application still trying to write to the Default URL(http://localhost:9200). 

I'm using CoreOS. And everything is works when I'm running Cadvisor on the machine with running local Elasticsearch container.

### PR DESCRIPTION
**Pull request by hacpai**
* fix #881 
* Add support for ElasticSearch heath check and find all nodes of cluster
* Rename command-line flag about ElasticSearch

### CODE REVIEW

File: storage/elasticsearch/elasticsearch.go [hunk 1]
```diff
@@ -93,19 +94,24 @@ func New(machineName,
 	indexName,
 	typeName,
 	elasticHost string,
+	enableSniffe bool,
```
Comment (vishh) [inline storage/elasticsearch/elasticsearch.go]: nit: `Sniff` or `Sniffer`

File: storagedriver.go [hunk 2]
```diff
@@ -39,9 +39,10 @@ var argDbTable = flag.String("storage_driver_table", "stats", "table name")
 var argDbIsSecure = flag.Bool("storage_driver_secure", false, "use secure connection with database")
 var argDbBufferDuration = flag.Duration("storage_driver_buffer_duration", 60*time.Second, "Writes in the storage driver will be buffered for this duration, and committed to the non memory backends as a single transaction")
 var storageDuration = flag.Duration("storage_duration", 2*time.Minute, "How long to keep data stored (Default: 2min).")
-var argElasticHost = flag.String("storage_driver_es_host", "http://localhost:9200", "database host:port")
-var argIndexName = flag.String("storage_driver_index", "cadvisor", "index name")
-var argTypeName = flag.String("storage_driver_type", "stats", "type name")
+var argElasticHost = flag.String("storage_driver_es_host", "http://localhost:9200", "ElasticSearch host:port")
+var argIndexName = flag.String("storage_driver_es_index", "cadvisor", "ElasticSearch index name")
+var argTypeName = flag.String("storage_driver_es_type", "stats", "ElasticSearch type name")
+var argEnableSniffe = flag.Bool("storage_driver_es_enable_sniffe", false, "ElasticSearch uses a sniffing process to find all nodes of your cluster by default, automatically")
```
Comment (vishh) [inline storagedriver.go]: Are you referring to elasticsearch finding its peers here?

**Comment by vishh** [comment]
ok to test

**Comment by vishh** [comment]
@hacpai: Thanks for the PR! AFAIK there are no integration tests for elasticsearch driver. It might be worth adding an integration test incase you rely on this feature to be stable.

File: storagedriver.go (hunk 2 above)
Comment (hacpai) [inline storagedriver.go]: Yes.

**Comment by vishh** [comment]
LGTM

**Pull request merged**